Source/WebCore/animation/ScrollTimeline.cpp

```diff
@@ -204,7 +204,14 @@ ScrollTimeline::Data ScrollTimeline::computeTimelineData(const TimelineRange& ra
     float maxScrollOffset = axis() == ScrollAxis::Block ? sourceScrollableArea->maximumScrollOffset().y() : sourceScrollableArea->maximumScrollOffset().x();
     float scrollOffset = axis() == ScrollAxis::Block ? sourceScrollableArea->scrollOffset().y() : sourceScrollableArea->scrollOffset().x();
 
-    return { scrollOffset, floatValueForOffset(range.start.offset, maxScrollOffset), maxScrollOffset - floatValueForOffset(range.end.offset, maxScrollOffset) };
+    // Chrome appears to clip the current time of a scroll timeline in the [0-100] range.
+    // We match this behavior for compatibility reasons, see https://github.com/w3c/csswg-drafts/issues/11033.
+    auto rangeStart = floatValueForOffset(range.start.offset, maxScrollOffset);
+    auto rangeEnd = maxScrollOffset - floatValueForOffset(range.end.offset, maxScrollOffset);
+    if (rangeEnd > rangeStart)
+        scrollOffset = std::clamp(scrollOffset, rangeStart, rangeEnd);
```

Comment:
This is not quite right. Currently, chrome clamps between 0 and maxScrollOffset, not the progress range. This is visible when there is an animation-range that shortens the progress range for the animation, which will cause the animation.currentTime to return a negative progress if the scroll position is before the active range.

Reply:
Right on, fixed in the latest iteration of this patch.